### PR TITLE
Refine the music playlist code.

### PR DESCRIPTION
**Pull request by pchote**
This implements the plan from #8746 with some adaptations.
Also fixes a crash when opening the music player from the TD lobby, and randomizes the default starting song so that it doesn't always play the first song in music.yaml if shuffle is disabled.

### CODE REVIEW

File: OpenRA.Game/Traits/World/MusicPlaylist.cs [hunk 1]
```diff
@@ -41,44 +36,57 @@ public class MusicPlaylistInfo : ITraitInfo
...
 		public MusicPlaylist(World world, MusicPlaylistInfo info)
 		{
 			this.info = info;
-
-			IsMusicAvailable = world.Map.Rules.InstalledMusic.Any();
+			this.world = world;
 
 			playlist = world.Map.Rules.InstalledMusic.Select(a => a.Value).ToArray();
 
+			IsMusicAvailable = playlist.Length > 1;
```
Comment (Phrohdoh) [inline OpenRA.Game/Traits/World/MusicPlaylist.cs]: Why is music not available if we only have 1 song?

Comment (pchote) [inline OpenRA.Game/Traits/World/MusicPlaylist.cs]: This was inherited from @obrakmann's original implementation.  I've removed it for now so we can argue about it in a followup PR.  See #8747.

**Comment by pchote** [comment]
I have scope-creeped this to fix the rest of our music issues too.

**Comment by GraionDilach** [comment]
Oh.... Oh.

What does `ShippedSoundtracks` check now? I might messed it up back then with the first music commit, via enabling the other two tracks from `transit.mix`.

On the other hand, now I can see what got me confused - I wouldn't have added the `BackgroundMusic` system separately.

Also really sorry that you had to do this. >_>

**Comment by Mailaender** [comment]
Seems to work as intended. :white_check_mark: Can't test d2k/ts due to https://github.com/OpenRA/OpenRA/issues/8884, but that didn't regress here.

**Comment by obrakmann** [comment]
Looks fine. :+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/d518fb3c6a845b82caf9157941ce06a9bb794b6a%5E...d518fb3c6a845b82caf9157941ce06a9bb794b6a)

**Pull request merged**